symbulate/probability_space.py

```diff
@@ -10,10 +11,16 @@ class ProbabilitySpace:
     Attributes:
       draw (function): A function explaining how to draw one 
         outcome from the probability space.
+      pf (function): A function giving the exact probability
+        density (if continuous) or mass (if discrete) from the probability space 
+        for a given input.
+      discrete (boolean): Describes if the probability space is discrete or not.
     """
 
-    def __init__(self, draw):
+    def __init__(self, draw, pf = None, discrete = False):
```

Comment:
Instead of adding pf and discrete to ProbabilitySpace class, create a Distribution subclass with these kwargs, and all the distributions will be instances of that. Only Distribution supports plotting, not ProbabilitySpace. I think you can revert probability_space.py to how it was.